# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsPIDOMWindow.h
# Commit: 0c5428075ca4
# Full Hash: 0c5428075ca4aa6b5270350247d43217fa5d0feb
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-07 08:31:23
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 9e7fe3dbea01 -r 0c5428075ca4 dom/base/nsPIDOMWindow.h
--- a/dom/base/nsPIDOMWindow.h	Fri Jun 06 14:38:54 2025 +0000
+++ b/dom/base/nsPIDOMWindow.h	Fri Jun 06 14:38:54 2025 +0000
@@ -66,6 +66,7 @@
 class ServiceWorkerDescriptor;
 class Timeout;
 class TimeoutManager;
+class WebIdentityHandler;
 class WindowContext;
 class WindowGlobalChild;
 class CustomElementRegistry;
@@ -621,6 +622,8 @@
 
   bool UsingStorageAccess();
 
+  mozilla::dom::WebIdentityHandler* GetOrCreateWebIdentityHandler();
+
   uint32_t UpdateLockCount(bool aIncrement) {
     MOZ_ASSERT_IF(!aIncrement, mLockCount > 0);
     mLockCount += aIncrement ? 1 : -1;
@@ -666,6 +669,7 @@
 
   RefPtr<mozilla::dom::Performance> mPerformance;
   mozilla::UniquePtr<mozilla::dom::TimeoutManager> mTimeoutManager;
+  RefPtr<mozilla::dom::WebIdentityHandler> mWebIdentityHandler;
 
   RefPtr<mozilla::dom::Navigation> mNavigation;
 